src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -2014,18 +2016,30 @@ public InternalPreferences getInternalPreferences() {
         return internalPreferences;
     }
 
-    private String getUser() {
-        if (StringUtil.isNotBlank(userName)) {
-            return userName;
+    String getUserAndHost() {
+        if (StringUtil.isNotBlank(userAndHost)) {
+            return userAndHost;
         }
 
-        try {
-            userName = get(DEFAULT_OWNER) + '-' + InetAddress.getLocalHost().getHostName();
-            return userName;
-        } catch (UnknownHostException ex) {
-            LOGGER.error("Hostname not found. Please go to https://docs.jabref.org/ to find possible problem resolution", ex);
-            return get(DEFAULT_OWNER);
+        String hostName;
+        // Following code inspired by https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/SystemUtils.html#getHostName--
+        // See also https://stackoverflow.com/a/20793241/873282
+        if (OS.WINDOWS) {
+            hostName = System.getenv("COMPUTERNAME");
+        } else {
+            hostName = System.getenv("HOSTNAME");
         }
+        if (StringUtil.isBlank(hostName)) {
+            try {
+                hostName = InetAddress.getLocalHost().getHostName();
+            } catch (UnknownHostException e) {
+                LOGGER.info("Hostname not found. Using \"localhost\" as fallback.", e);
+                hostName = "localhost";
+            }
+        }
+
+        userAndHost = get(DEFAULT_OWNER) + '-' + hostName;
+        return userAndHost;
     }
```

Comment:
Based on the open-closed design principle, i believe this should all go to the OS classes instead, so the JabRefPreferences does not have to distinguish between the different os, but just call to the proper overridden method. Also the call directly to `System.getenv` would imho break the level of abstraction.

Reply:
The change was easy: [`151ea22` (#9910)](https://github.com/JabRef/jabref/pull/9910/commits/151ea222da887d547f0415780b4e2318631fbf7c). Huge preparation work required though: [`84899aa` (#9910)](https://github.com/JabRef/jabref/pull/9910/commits/84899aae9028d506f09c869bd9a1bd7e6dc2316a).